src/urllib3/util/ssl_match_hostname.py

```diff
@@ -77,10 +85,10 @@ def _ipaddress_match(ipname, host_ip):
     # OpenSSL may add a trailing newline to a subjectAltName's IP address
     # Divergence from upstream: ipaddress can't handle byte str
     ip = ipaddress.ip_address(ipname.rstrip())
-    return ip == host_ip
+    return bool(ip == host_ip)
 
 
-def match_hostname(cert, hostname):
+def match_hostname(cert: _PeerCertRetType, hostname: str) -> None:
```

Comment:
Shouldn't `cert` be `_PeerCertRetDictType` here?

Reply:
I copied it from https://github.com/urllib3/urllib3/pull/2198/files#diff-e60901ca7e95a57497217ad9b1a4228bbe3319f6d264575a4f667ac349b015c7 as @pquentin suggested.